atlasaction/action.go

```diff
@@ -550,6 +567,72 @@ func (a *Actions) MigrateTest(ctx context.Context) error {
 	return nil
 }
 
+// MigrateAutoRebase runs the Action for "ariga/atlas-action/migrate/autorebase"
+func (a *Actions) MigrateAutoRebase(ctx context.Context) error {
+	dirpath := strings.TrimPrefix(a.GetInput("dir"), "file://")
+	rebaseBranch := a.GetInput("rebase-branch")
+	if rebaseBranch == "" {
+		rebaseBranch = "main"
+	}
+	tc, err := a.GetTriggerContext(ctx)
+	if err != nil {
+		return err
+	}
+	branch := tc.GetRunContext().Branch
+	if err := a.CmdExecutor(ctx, "git", "fetch", "origin", rebaseBranch).Run(); err != nil {
+		return fmt.Errorf("failed to fetch the branch %s: %w", rebaseBranch, err)
+	}
+	// Since running in detached HEAD, we need to switch to the branch.
+	if err := a.CmdExecutor(ctx, "git", "checkout", branch).Run(); err != nil {
+		return fmt.Errorf("failed to checkout to the branch: %w", err)
+	}
+	incoming, err := a.CmdExecutor(ctx, "git", "show", "origin/"+rebaseBranch+":"+dirpath+"/atlas.sum").Output()
+	if err != nil {
+		return fmt.Errorf("failed to get the atlas.sum file from the rebase branch: %w", err)
+	}
+	base, err := os.ReadFile(dirpath + "/atlas.sum")
+	if err != nil {
+		return fmt.Errorf("failed to read atlas.sum file: %w", err)
+	}
+	rebaseFiles := git.FilesOnlyInBase(string(base), string(incoming))
+	if len(rebaseFiles) == 0 {
+		a.Infof("No files to rebase")
+		return nil
+	}
+	if err := a.CmdExecutor(ctx, "git", "config", "--global", "user.name", "atlas-action").Run(); err != nil {
+		return fmt.Errorf("failed to set the git user name: %w", err)
+	}
+	if err := a.CmdExecutor(ctx, "git", "config", "--global", "user.email", "atlas-action@ariga").Run(); err != nil {
+		return fmt.Errorf("failed to set the git user email: %w", err)
+	}
```

Comment:
I prefer to have option to skip this. The user may already have the git account on their CI. This also override the git account of the user